src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,24 +1,86 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 /**
  * <p>Реалізувати свою HashMap, а саме методи `put(K key, V value)`, `getValue()` та `getSize()`.
  * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+
+    private Node<K, V>[] buckets;
+    private int size;
+    private float threshold;
+
+    public MyHashMap() {
+        buckets = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndex(key);
+        Node<K, V> node = buckets[index];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
+        Node<K, V> newNode = new Node<>(key, value);
+        newNode.next = buckets[index];
+        buckets[index] = newNode;
+        size++;
+        if (size >= buckets.length * LOAD_FACTOR) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> node = buckets[index];
+        while (node != null) {
+            if (node.key == key || node.key != null && node.key.equals(key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
+    private void resize() {
+        size = 0;
+        Node<K, V>[] oldBusckets = buckets;
+        buckets = new Node[buckets.length * 2];
+        for (Node<K, V> node : oldBusckets) {
+            while (node != null) {
+                put(node.key, node.value);
+                node = node.next;
+            }
+        }
+    }
+
+    private int getIndex(K key) {
+        return key == null ? 0 : ((key.hashCode() & (buckets.length - 1)));
+    }
+
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    static class Node<K, V> {
```

Comment:
what about access modifier, should it be default?

Reply:
changed to private )